### PR TITLE
BOAC-420, add/remove from watchlist on profile and list view

### PR DESCRIPTION
**Pull request by johncrossman**
https://jira.ets.berkeley.edu/jira/browse/BOAC-420
https://jira.ets.berkeley.edu/jira/browse/BOAC-417

UI samples:
image

image

Designs will likely evolve but the front-end wiring is in place.

### CODE REVIEW

File: boac/static/app/user/watchlistToggleDirective.js [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    };
+
+    return {
+      // @see https://docs.angularjs.org/guide/directive#template-expanding-directive
+      restrict: 'E',
+
+      // @see https://docs.angularjs.org/guide/directive#isolating-the-scope-of-a-directive
+      scope: {
+        sid: '=',
+        view: '=bind',
```
Comment (paulkerschen) [inline boac/static/app/user/watchlistToggleDirective.js]: @johncrossman Do you want `=bind` or `=` here? I'd expect `=bind` to look for an attribute called `data-bind` in the HTML.

Comment (johncrossman) [reply boac/static/app/user/watchlistToggleDirective.js]: @paulkerschen  Ah, I now see the prob I hit earlier. On landing page, for example, I pass 'student' expecting to get fa styling of profile page. However, I missed the fact that 'student' is defined in `$scope` and that causes a conflict of sorts. The `'=bind'` worked around the conflict w/o understanding the problem. (Note: use of 'bind' should not change attribute naming)

Comment (johncrossman) [reply boac/static/app/user/watchlistToggleDirective.js]: Amendment is on the way...

File: boac/static/app/user/watchlistToggleDirective.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+    return {
+      // @see https://docs.angularjs.org/guide/directive#template-expanding-directive
+      restrict: 'E',
+
+      // @see https://docs.angularjs.org/guide/directive#isolating-the-scope-of-a-directive
+      scope: {
+        sid: '=',
+        watchlist: '='
+      },
```
Comment (johncrossman) [inline boac/static/app/user/watchlistToggleDirective.js]: @paulkerschen Fix is in. I choose to go with more explicit naming of attribute ('faSize') and remove from `scope` above.

**Comment by paulkerschen** [comment]
Thanks @johncrossman!

**Pull request merged**